### PR TITLE
Support hl

### PR DESCRIPTION
**Pull request by sdashevsky-ai**

### CODE REVIEW

File: src/hl_memory.c [hunk 1]
```diff
@@ -6,101 +6,171 @@
...
+struct KeyValuePair {
+	uint64_t key;
+	uint64_t value;
+	bool is_occupied;
+};
 
 struct hl_memory_ctx {
 	struct memory_ctx base;
 	char *device_bus_id;
 	synDeviceId device_id;
```
Comment (imasuari) [inline src/hl_memory.c]: you are not making use of this member, can be removed?

File: src/hl_memory.c [hunk 2]
```diff
@@ -6,101 +6,171 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
+#include <pthread.h>
+#include <fcntl.h>
 #include "hl_memory.h"
 #include "perftest_parameters.h"
 #include "synapse_api.h"
```
Comment (imasuari) [inline src/hl_memory.c]: is here only for `synDeviceId`, if the member will be removed, remove this include as well

File: src/hl_memory.c [hunk 3]
```diff
@@ -6,101 +6,171 @@
...
+		(void) pthread_mutex_unlock(&hl_ctx->mem_handle_table_lock);
+		return FAILURE;
+	}
+	if (pthread_mutex_unlock(&hl_ctx->mem_handle_table_lock) != 0) {
+		fprintf(stderr, "Failed to unlock mutex\n");
 		return FAILURE;
 	}
 
-	fd = hlthunk_device_memory_export_dmabuf_fd(hl_ctx->device_fd, buffer_addr, buf_size, 0);
+	fd = hlthunk_device_mapped_memory_export_dmabuf_fd(hl_ctx->device_fd, buffer_addr, buf_size, NO_OFFSET,
```
Comment (imasuari) [inline src/hl_memory.c:123]: is this code supposed to be backward compatible to gaudi 1?
if so, this call was changed, need to call `hlthunk_device_memory_export_dmabuf_fd` if the device is gaudi 1

(very confusing names, hlthunk_device_memory_export_dmabuf_fd vs hlthunk_device_mapped_memory_export_dmabuf_fd)

**Review by imasuari — APPROVED** [review]

**Comment by sdashevsky-ai** [comment]
This change doesn't support gaudi 1 as it uses old synapse API to allocate memory and not hlthunk which is currently used.

**Pull request merged**